util/journaldb/src/overlayrecentdb.rs

```diff
@@ -70,12 +70,38 @@ use crate::{
 /// the removed key is not present in the history overlay.
 /// 7. Delete ancient record from memory and disk.
 
+const CACHE_MEM_USED_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(60);
+
+pub struct CachedMemUsed {
+	mem_used: usize,
+	ts: std::time::Instant,
+	outdated : bool,
+}
+impl CachedMemUsed {
+	pub fn new(mem_used: usize) -> Self {
+		Self {
+			mem_used,
+			ts : std::time::Instant::now(),
+			outdated : false,
+		}
+	}
+	pub fn value_or_none_once(&mut self) -> Option<usize> {
+		if !self.outdated && self.ts.elapsed() > CACHE_MEM_USED_TIMEOUT {
+			self.outdated = true;
```

Comment:
```suggestion
		if self.ts.elapsed() > CACHE_MEM_USED_TIMEOUT {
```
I don't think `Instant::elapsed()` is that expensive, therefore we might not need to cache it using `outdated`.

Reply:
The idea here is to prevent two threads from scanning the memory at the same time due to an outdated cache.  `None` is only returned once to signal the caller to re-scan the memory and refresh the cache.

Reply:
Yes, but the time only goes forward. Once `self.ts.elapsed() > CACHE_MEM_USED_TIMEOUT` starts yielding true, it will always do so and thus take the `None` route. We don't need to set an additional flag here.

Reply:
For instance we have two threads, with the following timing:

0:00s th1 mem_used
0:02s th2 mem_used

and the memory scanning takes 5s

```
th1                             th2
----------------------------    ----------------------------
mem_used
   value_or_none_once
       set outdated=1
       return None
   compute size_of               mem_used
       (5s...)                      value_of_nonce_once
                                      return Some(cached value)
                                    return cached value  
   update cache
       set outdated=0
   return new cache value
```

if I only put the `self.ts.elapsed() > CACHE_MEM_USED_TIMEOUT` cond:

```
th1                             th2
----------------------------    ----------------------------
mem_used
   value_or_none_once
       return None
   compute size_of               mem_used
       (5s...)                     value_or_nonce_once
                                         return None
                                   compute size_of
                                         (5s...)  
```